File: 250115/오목/O-mok.cpp
#include <iostream>
using namespace std;

int dx[8]={1, 1, 1, -1, -1, -1, 0, 0}, dy[8]={-1, 0, 1, -1, 0, 1, -1, 1};

bool InRange(int r, int c) {
    return ((1 <= r && r <= 19) && (1 <= c && c <= 19));
}

int main() {
    // Please write your code here.
    int arr[20][20];
    for (int i=1; i<=19; i++) {
        for (int j=1; j<=19; j++) {
            cin >> arr[i][j];
        }
    }
    for (int i=1; i<=19; i++) {
        for (int j=1; j<=19; j++) {
            if (arr[i][j] == 0) continue;
            for (int k=0; k<8; k++) {
                int cnt = 1;
                int x = i;
                int y = j;
                while (true) {
                    int nx = x + dx[k];
                    int ny = y + dy[k];
                    if (!InRange(nx, ny)) break;
                    if (arr[nx][ny] == 0) break;
                    x = nx;
                    y = ny;
                    cnt++;
                }    
                if (cnt == 5) {
                    cout << arr[i][j] << endl;
                    cout << i+ 2*dx[k] << ' ' << j+ 2*dy[k];
                    return 0;
                }
            }
        }
    }
    cout << 0;
    return 0;
}

File: 250630/오목/O-mok.py
def in_range(r, c):
    return (0 <= r and r < 19) and (0 <= c and c < 19)

arr = [list(map(int, input().split())) for i in range(19)]

dr = [-1, -1, -1, 0, 0, 1, 1, 1]
dc = [-1, 0, 1, -1, 1, -1, 0, 1]

for i in range(19):
    for j in range(19):
        if arr[i][j] == 0:
            continue

        cnt = 0
        for k in range(4):
            r, c = i, j
            for l in range(2):
                r += dr[k]
                c += dc[k]
                if in_range(r, c) and arr[r][c] == arr[i][j]:
                    cnt += 1

            r, c = i, j
            for l in range(2):
                r += dr[7-k]
                c += dc[7-k]
                if in_range(r, c) and arr[r][c] == arr[i][j]:
                    cnt += 1
            if cnt == 4:
                print(arr[i][j])
                print(i+1, j+1)
                exit()

print(0)
